### PR TITLE
Implement embed

### PR DESCRIPTION
**Pull request by NotFlameDev**

### CODE REVIEW

File: src/cordslash/models.py [hunk 1]
```diff
@@ -13,4 +17,85 @@ class Mentionable:
     ...
 
 
+class Embed:
+    def __init__(self, /, title: str = None,
```
Comment (an-dyy) [inline src/cordslash/models.py:21]: Inconsistency: `/` is used here and every other function that are `kwarg-only` use `*`

File: src/cordslash/models.py [hunk 2]
```diff
@@ -13,4 +17,85 @@ class Mentionable:
...
+            self.author["icon_url"] = icon_url
+
+    def set_footer(self, *, text, icon_url = None):
+        self.footer = {
+            "text": text
+        }
+        if icon_url:
+            self.footer["icon_url"] = icon_url
+
+    def set_image(self, *, url, height: int = None, width: int = None):
```
Comment (an-dyy) [inline src/cordslash/models.py:57]: You cannot set the height and width of an image

File: src/cordslash/models.py [hunk 3]
```diff
@@ -13,4 +17,85 @@ class Mentionable:
...
+        if height:
+            self.image["height"] = height
+        if width:
+            self.image["width"] = width
+
+    def set_thumbnail(self, *, url, height: int = None, width: int = None):
+        self.thumbnail = {
+            "url": url
+        }
+        if height:
```
Comment (an-dyy) [inline src/cordslash/models.py:70]: Same as the last comment

File: src/cordslash/models.py [hunk 4]
```diff
@@ -13,4 +17,85 @@ class Mentionable:
...
+
+    def __len__(self):
+        length = len(self.title) + len(self.description)
+        for field in self.fields:
+            length += len(field["name"]) + len(field["value"])
+        length += len(self.footer["text"]) + len(self.author["name"]) + len(self.timestamp)
+        return length
+
+
+    def from_dict(self, embed_dict: Union[dict, str]):
```
Comment (an-dyy) [inline src/cordslash/models.py:83]: I don't see why this method should take in both a `dict` and a `str`, the user should supply a dict

File: src/cordslash/models.py [hunk 5]
```diff
@@ -13,4 +17,85 @@ class Mentionable:
...
 
 
+class Embed:
+    def __init__(self, /, title: str = None,
+                 description: str = None,
+                 color: int = 0x000000,
+                 timestamp: Union[datetime.datetime, str] = None,
+                 url:str = None,
+                 type: str = "rich"):
+        self.title = str(title) if title else None
```
Comment (an-dyy) [inline src/cordslash/models.py:27]: No need for `str()` since these should all be strings

**Review by an-dyy — CHANGES_REQUESTED** [review]
Just some minor changes and this should be fine, I can handle the typehinting and formatting,.. *etc*

File: src/cordslash/models.py (hunk 5 above)
Comment (NotFlameDev) [reply src/cordslash/models.py:27]: hmm, the title str(title) should be fine, the problem is with the typehint which is str, as use can pass int, object, etc to title and description.

**Pull request merged**